CMV-14 test and management

FrancoisChastel opened the pull request.

FrancoisChastel committed:
Issue #13, feat: adding CMV-5 management

FrancoisChastel committed:
Issue #13, test: adding CMV-5 test

FrancoisChastel committed:
Issue #22, feat: adding CMV-14 management

FrancoisChastel committed:
Issue #22, test: adding CMV-14 test

FrancoisChastel committed:
Merge branch 'master' into CMV-issue14

FrancoisChastel committed:
Merge branch 'master' into CMV-issue14

JonasOlofssonKTH commented on Decide/src/main/java/DD2480/decide/CMVHandler.java:
```diff
@@ -253,7 +257,20 @@ private boolean licThirteen() {
-		// TODO Auto-generated method stub
-		return false;
-	}	
+        boolean greater = false;
```
You should check if the parameters are valid:

This most be true NUMPOINTS >= 5 and  0 < AREA2

JonasOlofssonKTH commented on Decide/src/main/java/DD2480/decide/CMVHandler.java:
```diff
@@ -253,7 +257,20 @@ private boolean licThirteen() {
+        boolean greater = false;
+        boolean lesser = false;
+
+        for (int index = 0; index < (parameters.kPts - parameters.ePts - parameters.fPts - 2); index++) {
```
change 
for (int index = 0; index < (parameters.kPts - parameters.ePts - parameters.fPts - 2); index++) {
to this
for (int index = 0; index < (dataPoints.length - parameters.ePts - parameters.fPts - 2); index++) {

JonasOlofssonKTH commented on Decide/src/test/java/DD2480/decide/CMVHandlerTest.java:
```diff
@@ -529,6 +565,59 @@ public void testCMV12nice(){
+    public void testCMV14bad() {
+        boolean[] CMV = new boolean[15];
+        Parameter parameters = new Parameter();
+        parameters.kPts = 5;
```
kPts is used in lic 7 and lic 12, not 14

JonasOlofssonKTH commented on Decide/src/test/java/DD2480/decide/CMVHandlerTest.java:
```diff
@@ -529,6 +565,59 @@ public void testCMV12nice(){
+    public void testCMV14nice() {
+        boolean[] CMV = new boolean[15];
+        Parameter parameters = new Parameter();
+        parameters.kPts = 5;
```
same as above

JonasOlofssonKTH committed:
(FIX: #22) added validation of parameters Lic14

Check if dataPoints.length < 5 || parameters.area2 < 0.

Also corrected for loop so that it loops through dataPoints instead of parameter.kPts

JonasOlofssonKTH committed:
(FIX: #22) removed parameters not used in lic14

Removed parameters.kPts because it should not be used in lic14.

fkam committed:
(fix) fixed merge conflicts

fkam committed:
Merge branch 'master' into CMV-issue14
